SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -203,21 +203,20 @@ void UGlobalStateManager::ApplyStartupActorManagerUpdate(const Worker_ComponentU
 {
 	Schema_Object* ComponentObject = Schema_GetComponentUpdateFields(Update.schema_type);
 
-	if (Schema_GetBoolCount(ComponentObject, SpatialConstants::STARTUP_ACTOR_MANAGER_CAN_BEGIN_PLAY_ID) == 1)
+	if (Schema_GetBoolCount(ComponentObject, SpatialConstants::STARTUP_ACTOR_MANAGER_AUTH_BEGIN_PLAY_CALLED_ID) == 1)
 	{
-		bool bCanBeginPlayUpdate = GetBoolFromSchema(ComponentObject, SpatialConstants::STARTUP_ACTOR_MANAGER_CAN_BEGIN_PLAY_ID);
-		ApplyCanBeginPlayUpdate(bCanBeginPlayUpdate);
+		const bool bAuthBeginPlayCalledUpdate = GetBoolFromSchema(ComponentObject, SpatialConstants::STARTUP_ACTOR_MANAGER_AUTH_BEGIN_PLAY_CALLED_ID);
+		ApplyAuthBeginPlayCalledUpdate(bAuthBeginPlayCalledUpdate);
 	}
 }
 
-void UGlobalStateManager::ApplyCanBeginPlayUpdate(bool bCanBeginPlayUpdate)
+void UGlobalStateManager::ApplyAuthBeginPlayCalledUpdate(const bool bAuthBeginPlayCalledUpdate)
 {
-	bCanBeginPlay = bCanBeginPlayUpdate;
+	bAuthBeginPlayCalled = bAuthBeginPlayCalledUpdate;
 
-	// For now, this will only be called on non-authoritative workers.
-	if (bCanBeginPlay)
+	if (!bIsReady)
 	{
-		TryTriggerBeginPlay();
+		bIsReady = true;
```

Comment:
bIsReady seems a little non-descript in the terms of the GSM. Do we actually need it? Couldn't we just check against the AuthBeginPlay bool?

Reply:
Could go with ReadyToCallBeginPlay?  So part of this change is that non-auth servers don't need to wait for AuthBeginPlay to be true to call their own BeginPlay, they just need to have received the op that updates AuthBeginPlay.

Reply:
But bIsReady is only going to be set to true after the authbeginplay op is processed right?

Reply:
yes